drools-ha/ha-core-infra/src/main/java/org/kie/hacep/core/infra/consumer/DefaultKafkaConsumer.java

```diff
@@ -93,28 +96,50 @@ public DefaultKafkaConsumer(EnvConfig config, Producer producer) {
         }
         this.producer = producer;
         this.consumerUtilsCore = new ConsumerUtilsCoreImpl();
+        this.snapshotOnDemandUtils = snapshotOnDemandUtils;
     }
 
-    public void initConsumer(ConsumerHandler consumerHandler) {
-        this.consumerHandler = (DroolsConsumerHandler) consumerHandler;
-        this.snapShooter = (DefaultSessionSnapShooter) InfraFactory.getSnapshooter(envConfig);
+    //For test
+    void setupForTest(EnvConfig config, Producer producer,
```

Comment:
Are you sure it is not possible to subclass/set directly some fields and avoid to have a test specific method?